### DISCUSSION TITLE
Making An Awesome Javascript Open Source Project

### DISCUSSION THREAD

**Discussion by ghost** [EddieHubCommunity/support#1726] [Show and tell] [answered]
## Code
- [ ] Linters: [ESlint](https://eslint.org/)
- [ ] Formatter: [Prettier](https://prettier.io/)
- [ ] Code Style: [StandardJs](https://standardjs.com/)

## GitHub
- [ ] Readme. Refer #1553
- [ ] Code Of Conduct. Recommendation: [this](https://www.contributor-covenant.org/version/2/0/code_of_conduct/code_of_conduct.md)
- [ ] Contributing.md. Include the code formatters and all guidelines
- [ ] Support.md. Refer [this](https://docs.github.com/en/communities/setting-up-your-project-for-healthy-contributions/adding-support-resources-to-your-project) 
- [ ] Issue Templates
       - [ ] **feat.md**: Feature Request
       - [ ] **docs.md**: Documentation Related Issue
       - [ ] **fix.md**: Bug Reports
       - [ ] [**config.yml**](https://docs.github.com/en/enterprise-server@3.0/communities/using-templates-to-encourage-useful-issues-and-pull-requests/configuring-issue-templates-for-your-repository#configuring-the-template-chooser): Add Your Chat Server/FAQ/Question Board
- [ ] Pull Request Templates
       - [ ] Use Check Marks For Indicating What All Has Been Completed
              Example:
              ```md
              - [ ] I have followed contributing Guidelines
              - [ ] I have followed coc
              - [ ] I have tested my code it doesn't break
              ```
- [ ] **Security.md** Tell people which versions are currently in support
       Example:
         ```md
         # Security Policy

       ## Supported Versions
       | Version | Supported |
       |------- | --------- |
       | 1.x.x   | ✅        |

       ## Reporting a Vulnerability

        To Report A Vulnerability. Mail The Author: **[krishdevdb@gmail.com](mailto:krishdevdb@gmail.com)**
         ```
- [ ] CI for code ⬆️ 

## Commits
- [ ] [Husky](https://typicode.github.io/husky/#/) - Lint your code automatically before `git commit`
- [ ] [Commitlint](https://commitlint.js.org/#/) - Check if your commits suit your commit convention
- [ ] [Conventional Commits](https://conventionalcommits.org/) - An Amazing Commit Style @eddiejaoude's favourite

> **Note All Of This Is Only For Code/Docs GitHub UI (Issues, PRs, Labels, Projects...) are not included**

**Comment by kaiwalyakoparkar** [comment]
Thanks for sharing :heart:

**Comment by Cahllagerfeld** [comment]
For planning and creating a large-scale JS-Project, I'd use Typescript to be honest. Thats a bit more work when setting up the project, but its more maintainable as the project grows

**Reply by ghost** [reply]
This one was just for javascript thus, typescript is not added, personally I also use ts

**Comment by ghost** [comment, marked as answer]
Up are the guidelines, if you have any suggestions please comment down, for now marking this as answered
